### PR TITLE
Enable strict-null-checks for api-extractor

### PR DESCRIPTION
**Pull request by pgonzal**
Turn on the **strict-null-checks** setting in tsconfig.json.

### CODE REVIEW

File: apps/api-extractor/src/aedoc/ApiDocumentation.ts [hunk 1]
```diff
@@ -377,8 +377,12 @@ export default class ApiDocumentation {
    * ensures that the reference is to an API item that is not 'Internal'.
    */
   private _completeLinks(): void {
-    while (this.incompleteLinks.length) {
-      const codeLink: IMarkupApiLink = this.incompleteLinks.pop();
+    for ( ; ; ) {
```
Comment (qz2017) [inline apps/api-extractor/src/aedoc/ApiDocumentation.ts]: >for ( ; ; ) [](start = 4, length = 11)

This is dangerous. Can you add an end condition?

File: apps/api-extractor/src/aedoc/ApiDocumentation.ts [hunk 2]
```diff
@@ -411,8 +415,12 @@ export default class ApiDocumentation {
    * for all API items.
    */
   private _completeInheritdocs(warnings: string[]): void {
-    while (this.incompleteInheritdocs.length) {
-      const token: Token = this.incompleteInheritdocs.pop();
+    for ( ; ; ) {
```
Comment (qz2017) [inline apps/api-extractor/src/aedoc/ApiDocumentation.ts]: >for ( ; ; ) [](start = 4, length = 11)

Same.

File: apps/api-extractor/src/aedoc/ApiDocumentation.ts (hunk 1 above)
Comment (pgonzal) [reply apps/api-extractor/src/aedoc/ApiDocumentation.ts]: The end condition is on line 383.

I can change it to `while (true)` if you don't like the `for (;;)` notation.  I will need to disable a tslint rule.

---
In reply to: [151211677](https://github.com/Microsoft/web-build-tools/pull/442#discussion_r151211677) [](ancestors = 151211677)

**Comment by pgonzal** [comment]
I validated that all the sp-client projects build with these api-extractor changes.

**Pull request merged**